lib/features/on_boarding/pages/onboarding_phone_page.dart

```diff
@@ -0,0 +1,141 @@
+  late final TextEditingController _controller;
+  final MaskTextInputFormatter maskFormatter = MaskTextInputFormatter(
+    mask: '(##) #####-####',
+  );
+  final SignupCubit _signupCubit = getIt.get();
+  bool canContinue = false;
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  // TODO(Natanael)
+  final int max = 15;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = TextEditingController();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<SignupCubit, SignupState>(
+      bloc: _signupCubit,
+      listener: (context, state) {
+        if (state.phoneIsValid) {
+          print(state.toString());
+          _auth.createUserWithEmailAndPassword(
+              email: state.email, password: state.password);
+          Navigator.pushReplacementNamed(context, AppRoutes.home);
+        }
+      },
+      builder: (_, state) {
+        return BaseOnBoardingPage(
+          content: Column(
+            children: [
+              DSTextField(
+                controller: _controller,
+                label: S.current.on_boarding_phone_page_title,
+                onChanged: (val) {
+                  if (val.length < max && canContinue == true) {
+                    setState(() {
+                      canContinue = false;
+                    });
+                  }
+                  if (val.length == max) {
+                    setState(() {
+                      canContinue = true;
+                    });
+                  }
+                },
+                maxLength: max,
+                keyboardType: TextInputType.number,
+                inputFormatters: [
+                  FilteringTextInputFormatter.digitsOnly,
+                  maskFormatter,
+                ],
+              ),
+              const VerticalGap.xxxs(),
+              StyledText(
+                text: S.current.on_boarding_phone_page_description,
+                textAlign: TextAlign.justify,
+                style: Theme.of(context).textTheme.bodyMedium,
+              ),
+            ],
+          ),
+          buttonEnabled: canContinue,
+          buttonLabel: S.current.continue_button_label,
+          onPressed: () async {
+            await _savePhoneNumber();
+            // Navigator.of(context).pushNamed(
+            //   AppRoutes.signup,
+            // );
+          },
+        );
+      },
+    );
+  }
+
+  Future<bool> _phoneNumberIsValid(String phoneNumber) async {
+    return await PhoneNumberUtil().validate(
+      phoneNumber,
+      regionCode: kRegion.code,
+    );
+  }
+
+  Future<void> _savePhoneNumber() async {
+    final rawPhoneNumber = maskFormatter.getUnmaskedText();
+    final phoneIsValid = await _phoneNumberIsValid(rawPhoneNumber);
+    if (phoneIsValid) {
+      _signupCubit.validatePhone(rawPhoneNumber);
+    } else {
```

Comment:
if phone is valid why validate again?

Reply:
function name is incorret it should be savePhone